discord-bot/src/index.ts

```diff
@@ -0,0 +1,133 @@
+      'Last week evm-faucet requests',
+      buildSlackStatsMessage('weekRecap', faucetRequestsCount),
+    )
+    faucetRequestsCount = 0
+  } else {
+    slackMessageId = await sendSlackMessage(
+      'Last week evm-faucet requests',
+      buildSlackStatsMessage('update', faucetRequestsCount),
+      slackMessageId,
+    )
+  }
+}
+
+// Only send a slack message if the last one was sent more than 1 hour ago
+const sendLowBalanceWarning = async (faucetBalance: BigNumber) => {
+  if (lastLowWarningMessage === undefined || new Date().getTime() - lastLowWarningMessage.getTime() > 60 * 60 * 1000) {
+    lastLowWarningMessage = new Date()
+    await faucetBalanceLowSlackMessage(utils.formatEther(faucetBalance))
+  }
+}
+
+//listens when bot is ready
+client.on('ready', (c) => {
+  log(`${c.user.tag} is online.`)
+})
+
+client.on('messageCreate', (message) => {
+  //check if person is bot, if yes do nothing
+  if (message.author.bot) return
+})
+
+//listen to interactions/event listener
+client.on('interactionCreate', async (interaction) => {
+  const member = interaction.member
+
+  if (!interaction.isChatInputCommand()) return
+
+  try {
+    if (interaction != null)
+      switch (interaction.commandName) {
+        case CommandNames.REQUEST_TOKENS: {
+          await interaction.deferReply()
+          log('\x1b[33m%s\x1b[0m', CommandNames.REQUEST_TOKENS, '\x1b[0m')
+          const addressOption = interaction.options.get('address')
+          if (addressOption === null || !addressOption.value || typeof addressOption.value !== 'string') {
+            interaction.followUp('Please provide an address')
+            return
+          } else {
+            const currentTime = BigNumber.from(Math.floor(Date.now() / 1000))
+            const nextAccessTime = await queries.nextAccessTime(addressOption.value)
+
+            // if nextAccessTime is bigger than currentTime, then the user has to wait
+            if (currentTime.gte(nextAccessTime)) {
+              const withdrawalAmount = await queries.withdrawalAmount()
+              const faucetBalance = await queries.verifyFaucetBalance()
+
+              // if faucetBalance is lower than withdrawalAmount * SLACK_BALANCE_NOTIFICATION_THRESHOLD, then send a slack message
+              if (faucetBalance.lt(withdrawalAmount.mul(BigNumber.from(config.SLACK_BALANCE_NOTIFICATION_THRESHOLD))))
+                await sendLowBalanceWarning(faucetBalance)
+
+              // if faucetBalance is lower than withdrawalAmount, then the faucet needs to be refilled
+              if (faucetBalance.lt(withdrawalAmount)) {
+                interaction.followUp('Faucet balance is too low, please wait for refill')
+                return
+              }
+              const formattedAmount = utils.formatEther(withdrawalAmount)
+              log('withdrawalAmount', formattedAmount)
+
+              // request tokens
+              const tx = await transactions.requestTokens(addressOption.value)
```

Comment:
Left a comment at utils about transaction, I'd personally prefer a direct call to `requestTokens` here.